### DISCUSSION TITLE
pyproject.toml seems to be ignored

### DISCUSSION THREAD

**Discussion by dstromberg** [astral-sh/ruff#6837] [Q&A] [answered]
Hi.

I have:

> $ cat ~/.config/ruff/pyproject.toml 
> below cmd output started 2023 Wed Aug 23 09:38:19 PM PDT
> [tool.ruff]
> line-length = 132
> 
> above cmd output done    2023 Wed Aug 23 09:38:20 PM PDT

...on a Debian 12 system running lunarvim (nvim 0.8.2).

But I'm finding that my preferred line length is no longer being respected by ruff.  It used to work well, and continues to work on some of my other machines.  I've been configuring them all using https://stromberg.dnsalias.org/svn/lunarvim-config/trunk

I have ruff hung off of pylsp.  ruff is reporting errors and warnings, but it's ignoring the line-length setting.

I also tried a ruff.toml in the same directory, but it didn't seem to help.

I don't think I have any pyproject.toml files in my project's working directory or any of its parent directories.

Any suggestions?

Thanks!

**Comment by dstromberg** [comment]
BTW, I know it's helpful to include a version of ruff in a problem report like this, but I can't seem to find the ruff binary on my system - even though I'm getting good ruff analysis inside lunarvim.  I tried a 'find' command like:
find / -xdev -name ruff -print
...but I didn't see it anywhere.

**Comment by charliermarsh** [comment]
Is there any chance that you have your LSP configured to pass in a line-length as a command-line argument? Ruff would always respect the command-line argument over any other settings. Apart from that, it'd be really useful to try and figure out the exact command that's being run by the LSP, in the logs or elsewhere, since that would presumedly also include the full path to the executable. If we know the exact command, we could run it with `--show-settings` to understand what settings are being inferred and why.

**Comment by dstromberg** [comment]
I checked TroubleToggle, and it looks like I'm not getting anything from ruff.  Maybe it's not even running.

**Reply by dstromberg** [reply, marked as answer]
It's not a ruff problem :)  It's really pycodestyle.  Thanks!

BTW, TroubleToggle is about indispensable.

**Reply by charliermarsh** [reply]
Woo, thanks for following up here :)
